src/hivpy/sexual_behaviour.py

```diff
@@ -1,35 +1,67 @@
+        return matrix_list[np.random.choice(matrix_list.shape[0])]
+
     def __init__(self, **kwargs):
         # Randomly initialise sexual behaviour group transitions
-        self.sex_behaviour_trans_male = np.random.choice(sb.sex_behaviour_trans_male_options)
-        self.sex_behaviour_trans_female = np.random.choice(sb.sex_behaviour_trans_female_options)
+        self.sex_behaviour_trans = {SexType.Male:
+                                    self.select_matrix(sb.sex_behaviour_trans_male_options),
+                                    SexType.Female:
+                                    self.select_matrix(sb.sex_behaviour_trans_female_options)}
         self.baseline_risk = sb.baseline_risk  # Baseline risk appears to only have one option
-        self.sex_mixing_matrix_female = np.random.choice(sb.sex_mixing_matrix_female_options)
-        self.sex_mixing_matrix_male = np.random.choice(sb.sex_mixing_matrix_male_options)
-        self.short_term_partners_male = np.random.choice(sb.short_term_partners_male_options)
-        self.short_term_partners_female = np.random.choice(sb.short_term_partners_female_options)
+        self.sex_mixing_matrix = {SexType.Male:
+                                  self.select_matrix(sb.sex_mixing_matrix_male_options),
+                                  SexType.Female:
+                                  self.select_matrix(sb.sex_mixing_matrix_female_options)}
+        self.short_term_partners = {SexType.Male:
+                                    self.select_matrix(sb.short_term_partners_male_options),
+                                    SexType.Female:
+                                    self.select_matrix(sb.short_term_partners_female_options)}
+
+    # Haven't been able to locate the probabilities for this yet
+    # Doing them uniform for now
+    def init_sex_behaviour_groups(self, population):
+        population["sex_behaviour"] = np.where(population['sex'] == SexType.Male,
+                                               np.random.choice(MaleSexBehaviour).value,
+                                               np.random.choice(FemaleSexBehaviour).value)
 
     # Here we need to figure out how to vectorise this which is currently blocked
-    # by the gender if statement
-    def prob_transition(self, gender, age, i, j):
+    # by the sex if statement
+    def prob_transition(self, sex, age, i, j):
         """Calculates the probability of transitioning from sexual behaviour
-        group i to group j, based on gender and age."""
-        if(gender == 1):
-            transition_matrix = self.sex_behaviour_trans_female
-            gender_index = 1
-        else:
-            transition_matrix = self.sex_behaviour_trans_male
-            gender_index = 0
+        group i to group j, based on sex and age."""
+        transition_matrix = self.sex_behaviour_trans[sex]
 
-        age_index = min((int(age)-15)//5, 9)
+        age_index = np.minimum((age.astype(int)-15)//5, 9)
```

Comment:
Am I right that the 9 here is because the `baseline_risk` matrix has 10 rows, to avoid indexing outside its bounds? If so, better avoid hardcoding it. Could have a field on the object that stores `self.baseline_risk.shape[1]` to avoid recomputing it.